test/e2e/tests/test_firewall.py

```diff
@@ -122,6 +122,65 @@ def create_adopted_resource_firewall(firewall_name):
 @service_marker
 @pytest.mark.canary
 class TestNetworkFirewall:
+    def test_create_delete(self, networkfirewall_client, simple_firewall):
+        (fw, cr) = simple_firewall
+
+        firewall_config = cr["status"]["firewall"]
+        firwall_name = firewall_config["firewallName"]
+
+        networkfirewall_validator = NetworkFirewallValidator(networkfirewall_client)
+        # Check Network Firewall comes to READY state
+        networkfirewall_validator.wait_for_firewall_creation_or_die(firwall_name, "READY", TIMEOUT_SECONDS)
+
+        # Check Network Firewall logging configuration exist in AWS
+        networkfirewall_validator.assert_network_firewall_logs(firwall_name, 1)
+
+        # Remove Network Firewall logging configuration
+        updates = {
+            "spec": {
+                "loggingConfiguration": None
+            }
+        }
+
+        k8s.patch_custom_resource(fw, updates)
+        time.sleep(UPDATE_WAIT_AFTER_SECONDS)
+        networkfirewall_validator.assert_network_firewall_logs(firwall_name, 0)
+
+        # Update Network Firewall logging configuration
+        resources = get_bootstrap_resources()
+        updates = {
+            "spec": {
+                "loggingConfiguration": {
+                    "logDestinationConfigs": [
+                        {
+                            "logDestination": {
+                                "bucketName": resources.LogsBucket.name
+                            },
+                            "logDestinationType": "S3",
+                            "logType": "ALERT"
+                        },
+                        {
+                            "logDestination": {
+                                "bucketName": resources.LogsBucket.name
+                            },
+                            "logDestinationType": "S3",
+                            "logType": "FLOW"
+                        }
+                    ]
+                }
+            }
+        }
+
+        k8s.patch_custom_resource(fw, updates)
+        time.sleep(UPDATE_WAIT_AFTER_SECONDS)
+        networkfirewall_validator.assert_network_firewall_logs(firwall_name, 2)
```

Comment:
Can we also test deletion and updates?

Reply:
Line 138 is testing the deletion of the logging configuration.
Line 149 is testing the update of the logging configuration.

Did you mean something else?